lightning/src/ln/onchaintx.rs

```diff
@@ -709,4 +775,44 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 			self.pending_claim_requests.remove(&req);
 		}
 	}
+
+	pub(super) fn provide_latest_local_tx(&mut self, tx: LocalCommitmentTransaction) -> Result<(), ()> {
+		// To prevent any unsafe state discrepancy between offchain and onchain, once local
+		// commitment transaction has been signed due to an event (either block height for
+		// HTLC-timeout or channel force-closure), don't allow any further update of local
+		// commitment transaction view to avoid delivery of revocation secret to counterparty
+		// for the aformentionned signed transaction.
+		if let Some(ref local_commitment) = self.local_commitment {
+			#[cfg(not(test))] //Test framework may need to generate revoked commitment tx, don't generate a monitor error
```

Comment:
Oof, I strongly disagree with this - this is an error condition that we should be writing tests to trigger, not turning off in testing. I really think the previous design of an exposed test-only version is much better.